File: engineering/codebase/smartqr.frontend-services/src/domain/codes/content/ContentMode.ts
/** Defines how a code carries its payload — baked directly (static) vs via the forwarder short link (dynamic). */
export const ContentMode = {
  /** Refers to a code that bakes the payload directly into the symbol (no redirect). */
  Static: "static",

  /** Refers to a code that carries the forwarder short link, resolved at scan time. */
  Dynamic: "dynamic",
} as const;

export type ContentMode = (typeof ContentMode)[keyof typeof ContentMode];

File: engineering/codebase/smartqr.backend-services/SmartQr.Common.Domain/Codes/Core/Enums/ContentMode.cs
namespace SmartQr.Common.Domain.Codes.Core.Enums;

/// <summary>Defines how a code's symbol resolves.</summary>
public enum ContentMode
{
    /// <summary>Represents a symbol carrying the payload itself.</summary>
    Static,

    /// <summary>Represents a symbol carrying the short link.</summary>
    Dynamic,
}
